block-storage/src/main/scala/coop/rchain/blockstorage/BlockStore.scala

```diff
@@ -15,6 +15,8 @@ trait BlockStore[F[_]] {
 
   def get(blockHash: BlockHash): F[Option[BlockMessage]]
 
+  def find(p: BlockHash => Boolean): F[Seq[(BlockHash, BlockMessage)]]
```

Comment:
Is it possible for the result to contain a hash that was not already supplied as parameter? If not - why return it?

Reply:
I would say that it's deprecated for external use. after it's not needed I would make it package private and let it be for tests. or we can try to replace it with `find`. I'm fine with that also.